### PR TITLE
Allow WithTx to Use Contexts

### PR DESCRIPTION
**Pull request by FahadBSyed**

### CODE REVIEW

**Review by albscui — APPROVED** [review]

File: src/internal/log/context.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 
 type pachydermLogger struct{}
 
+type Context context.Context
```
Comment (brendoncarroll) [inline src/internal/log/context.go:20]: Remove

File: src/internal/ppsutil/util.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func SetPipelineState(ctx context.Context, db *pachsql.DB, pipelinesCollection c
 	logSetPipelineState(ctx, pipeline, from, to, reason)
 	var resultMessage string
 	var warn bool
-	err := dbutil.WithTx(ctx, db, func(sqlTx *pachsql.Tx) error {
+	err := dbutil.WithTx(ctx, db, func(_ context.Context, sqlTx *pachsql.Tx) error {
```
Comment (brendoncarroll) [inline src/internal/ppsutil/util.go:173]: 👍

File: src/server/pfs/iface.go [hunk 3]
```diff
@@ -13,7 +15,7 @@ type APIServer interface {
 
 	NewPropagater(*txncontext.TransactionContext) txncontext.PfsPropagater
 
-	CreateRepoInTransaction(*txncontext.TransactionContext, *pfs_client.CreateRepoRequest) error
+	CreateRepoInTransaction(context.Context, *txncontext.TransactionContext, *pfs_client.CreateRepoRequest) error
```
Comment (brendoncarroll) [inline src/server/pfs/iface.go:18]: Doesn't the `TransactionContext` have a `context.Context` in it already?

File: src/server/pfs/server/driver.go [hunk 4]
```diff
@@ -122,7 +123,7 @@ func newDriver(env Env) (*driver, error) {
 	return d, nil
 }
 
-func (d *driver) createRepo(txnCtx *txncontext.TransactionContext, repo *pfs.Repo, description string, update bool) error {
+func (d *driver) createRepo(ctx context.Context, txnCtx *txncontext.TransactionContext, repo *pfs.Repo, description string, update bool) error {
```
Comment (brendoncarroll) [inline src/server/pfs/server/driver.go:126]: See my previous comment.  I don't think we need to add another context to all of these.  We should keep using the one from the `TransactionContext`

File: src/server/pfs/iface.go (hunk 3 above)
Comment (albscui) [reply src/server/pfs/iface.go:18]: We recently removed it.

**Pull request merged**